plugins/hls-cabal-project-plugin/test/Main.hs

```diff
@@ -0,0 +1,111 @@
+{-# LANGUAGE CPP                      #-}
+{-# LANGUAGE DisambiguateRecordFields #-}
+{-# LANGUAGE OverloadedStrings        #-}
+
+module Main (
+    main,
+) where
+
+import qualified Control.Exception             as E
+import           Control.Lens                  ((^.))
+import qualified Data.ByteString               as BS
+import           Data.ByteString.Char8         (pack)
+import           Data.Either                   (isRight)
+import qualified Ide.Plugin.CabalProject.Parse as Lib
+import qualified Language.LSP.Protocol.Lens    as L
+import           System.FilePath
+import           Test.Hls
+import           Utils
+
+
+main :: IO ()
+main = do
+    defaultTestRunner $
+        testGroup
+            "Cabal.project Plugin Tests"
+            [ unitTests
+            , pluginTests
+            ]
+
+-- ------------------------------------------------------------------------
+-- Unit Tests
+-- ------------------------------------------------------------------------
+
+unitTests :: TestTree
+unitTests =
+    testGroup
+        "Unit Tests"
+        [ cabalProjectParserUnitTests
+        ]
+
+cabalProjectParserUnitTests :: TestTree
+cabalProjectParserUnitTests =
+    testGroup
+        "Parsing Cabal.project"
+        [ testCase "Simple Parsing works" $ do
+            let fp    = testDataDir </> "cabal.project"
+            bytes <- BS.readFile fp
+            (warnings, pm) <- Lib.parseCabalProjectFileContents fp bytes
+            liftIO $ do
+                null warnings @? "Found unexpected warnings"
+                isRight pm @? "Failed to parse base cabal.project file"
+          , testCase "Correct root directory" $ do
+              let root    = testDataDir </> "root-directory"
+              let cabalFp = root </> "cabal.project"
+              bytes <- BS.readFile cabalFp
+              result <- E.try @E.IOException (Lib.parseCabalProjectFileContents cabalFp bytes)
+              case result of
+                Left err ->
+                  let errStr = show err
+                  in  (pack root `BS.isInfixOf` pack errStr)
+                        @?  ("Expected missing file error to mention the test-dir:\n"
+                        ++ "  " ++ root ++ "\n"
+                        ++ "but got:\n" ++ errStr)
+                Right _ ->
+                  False @? "Expected parse to fail (missing import), but it succeeded"
+        ]
+
+-- ------------------------ ------------------------------------------------
```

Comment:
I know it was like that, from where you copied it but please remove the space :upside_down_face: